src/debugger/Debugger.cpp

```diff
@@ -166,20 +166,22 @@ bool Debugger::doEval(ExecutionState* state, ByteCodeBlock* byteCodeBlock, uint8
     uint8_t type = (uint8_t)(buffer[0] + 1);
     uint32_t size;
     char* data;
+    uint32_t index;
 
     memcpy(&size, buffer + 1, sizeof(uint32_t));
+    memcpy(&index, buffer + 1 + sizeof(uint32_t), sizeof(uint32_t));
 
     if (size > ESCARGOT_DEBUGGER_MAX_MESSAGE_LENGTH - 1 - sizeof(uint32_t)) {
         if (length != ESCARGOT_DEBUGGER_MAX_MESSAGE_LENGTH) {
             goto error;
         }
 
-        data = (char*)GC_MALLOC_ATOMIC(size);
+        data = (char*)GC_MALLOC_ATOMIC(size - sizeof(uint32_t));
         char* ptr = data;
-        char* end = data + size;
+        char* end = data + size - sizeof(uint32_t);
 
-        length = ESCARGOT_DEBUGGER_MAX_MESSAGE_LENGTH - 1 - sizeof(uint32_t);
-        memcpy(ptr, buffer + 1 + sizeof(uint32_t), length);
+        length = ESCARGOT_DEBUGGER_MAX_MESSAGE_LENGTH - 1 - sizeof(uint32_t) - sizeof(uint32_t);
+        memcpy(ptr, buffer + 1 + sizeof(uint32_t) + sizeof(uint32_t), length);
```

Comment:
To simplify source sending, maybe we could read the whole data, and get the index later (similar to extracting the file name in source sending). But if this is not a problem, just disregard this comment.